amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -571,7 +571,8 @@ protected ShardConsumer buildConsumer(@NonNull final ShardInfo shardInfo,
                 shardDetector,
                 metricsFactory,
                 aggregatorUtil);
-        return new ShardConsumer(cache, executorService, shardInfo, lifecycleConfig.logWarningForTaskAfterMillis(), argument);
+        return new ShardConsumer(cache, executorService, shardInfo, lifecycleConfig.logWarningForTaskAfterMillis(),
+                argument, lifecycleConfig.taskExecutionListener());
```

Comment:
Add the listener to the ShardConsumerArgument. The argument was created for such scenarios where if new parameters need to be added  they can be.

Reply:
The `ShardConsumerArgument` was primarily added for things that are passed downstream to the tasks themselves, and not used by the `ShardConsumer`.  I think it's fine that the `TaskExecutionListener` is passed in, as it's used primarily by the ShardConsumer.